### PR TITLE
Doc update for Build 2020

### PR DESCRIPTION
**Pull request by faxu**
Incorporates changes for Training support

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -65,6 +94,8 @@ The shared library in the release Nuget(s) and the Python wheel may be installed
 |Ubuntu 16.x  | YES          | YES         | Also supported on ARM32v7 (experimental) |
 |Mac OS X  | YES          | NO         |    |
 
+* Red Hat Enterprise Linux and CentOS are not supported.
```
Comment (snnn) [inline BUILD.md:97]: "Red Hat Enterprise Linux and CentOS are not supported." is outdated and can be removed?

File: BUILD.md [hunk 2]
```diff
@@ -89,41 +121,30 @@ For other system requirements and other dependencies, please see [this section](
...
 |**Build Python wheel**|--build_wheel||
 |**Build C# and C packages**|--build_csharp||
-|**Build WindowsML**|--use_winml<br>--use_dml<br>--build_shared_lib|WindowsML depends on DirectML and the OnnxRuntime shared library.|
-|**Build Java package**|--build_java|Creates an onnxruntime4j.jar in the build directory, implies `--build_shared_lib`|
 
+---
 
-# Additional Build Instructions
-The complete list of build options can be found by running `./build.sh (or .\build.bat) --help`
+## Build ONNX Runtime Server on Linux
```
Comment (snnn) [inline BUILD.md:127]: Sorry, [this](https://github.com/microsoft/onnxruntime/blob/master/docs/Server.md) is the latest build doc for ORT Server.

File: BUILD.md [hunk 3]
```diff
@@ -158,7 +179,7 @@ A Dockerfile is available [here](./dockerfiles#cuda).
     * To use the 14.11 toolset with a later version of Visual Studio 2017 you have two options:
      1. Setup the Visual Studio environment variables to point to the 14.11 toolset by running vcvarsall.bat, prior to running the build script. e.g. if you have VS2017 Enterprise, an x64 build would use the following command `"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" amd64 -vcvars_ver=14.11` For convenience, .\build.amd64.1411.bat will do this and can be used in the same way as .\build.bat. e.g. ` .\build.amd64.1411.bat --use_cuda`
 
-     2. Alternatively, if you have CMake 3.13 or later you can specify the toolset version via the `--msvc_toolset` build script parameter. e.g. `.\build.bat --msvc_toolset 14.11`
+     2. Alternatively, if you have CMake 3.12 or later you can specify the toolset version via the `--msvc_toolset` build script parameter. e.g. `.\build.bat --msvc_toolset 14.11`
```
Comment (snnn) [inline BUILD.md:182]: Why do you change 3.13 to 3.12?

File: BUILD.md [hunk 4]
```diff
@@ -173,21 +194,15 @@ See more information on the TensorRT Execution Provider [here](./docs/execution_
 
 #### Pre-Requisites
 * Install [CUDA](https://developer.nvidia.com/cuda-toolkit) and [cuDNN](https://developer.nvidia.com/cudnn)
-   * The TensorRT execution provider for ONNX Runtime is built and tested with CUDA 10.2 and cuDNN 7.6.5.
+   * The TensorRT execution provider for ONNX Runtime is built and tested with CUDA 10.1 and cuDNN 7.6.
```
Comment (snnn) [inline BUILD.md:197]: ????

File: BUILD.md (hunk 1 above)
Comment (faxu) [reply BUILD.md]: Sorry, I had an outdated version. Fixed issues in latest commit.

File: BUILD.md (hunk 3 above)
Comment (faxu) [reply BUILD.md]: Sorry, I had an outdated version. Fixed issues in latest commit.

File: BUILD.md (hunk 4 above)
Comment (faxu) [reply BUILD.md]: Sorry, I had an outdated version. Fixed issues in latest commit.

File: BUILD.md (hunk 2 above)
Comment (faxu) [reply BUILD.md:129]: Thanks. Updated.

**Comment by pranavsharma** [comment]
Is the Eigen commit id update intentional?

**Comment by faxu** [comment]
> Is the Eigen commit id update intentional?

No. I think it was automatically updated locally. How do I revert this?

**Comment by codemzs** [comment]
@faxu Can you please try rebasing?

**Comment by faxu** [comment]
Closing PR. Will open a cleaner one.

**Pull request closed without merging**